atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricNameUtils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.util;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Preconditions;
+
+public final class MetricNameUtils {
+
+    private static final int MAX_ALLOWED_TAGS = 100;
+
+    private MetricNameUtils() {
+        //utility class
+    }
+
+    public static final String DELIMITER = ";";
+    public static final String EQUALS = "=";
+
+    public static String getMetricName(String metricName, Map<String, String> tags) {
```

Comment:
might want to also validate that `metricName` doesn't contain the DELIMITER, so we don't mess up when parsing out the tags

Reply:
Agreed, also the tag keys or values shouldn't contain the delimiter or the equals sign.

Reply:
Done.